text/0000-sandbox-environment.md

```diff
@@ -0,0 +1,201 @@
+- Feature Name: sandbox-environment
+- Start Date: 2019-10-26
+- RFC PR: [rust-lang/rfcs#2794](https://github.com/rust-lang/rfcs/pull/2794)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+This proposes a mechanism to precisely control what process environment is available to Rust programs at compilation time.
+
+# Motivation
+[motivation]: #motivation
+
+Rust supports the `env!` and `option_env!` pseudo-macros which allow Rust programs to query arbitrary process environment
+variables at compilation time. This is a very flexible mechanism to pass compile-time information to the program.
+
+However, in many cases it is too flexible. It poses several problems:
+1. Environment variables are generally not tracked by build systems, so changing a variable is not taken into account.
+   Cargo has an ad-hoc mechanism for doing this in build scripts, but there's nothing to make this guaranteed correct
+   (i.e. that all variables accessed are tracked).
+2. There's no easy way to audit which environment variables a process accesses. This not only exacerbates the problem
+   above, but it also means that potentially sensitive information in an environment variable can be incorporated into
+   the compiled code.
+3. There's no way to override variables if they're needed by the build process itself. For example, the `PATH` variable
+   likely needs to be set so that the compiler can execute its various components, but there's no way to override this
+   so that `env!("PATH")` returns something else. This would be necessary where the compilation environment differs from the
+   deployment environment (such as when cross-compiling).
+
+This RFC proposes a way to precisely control the environment visible to the compile-time macros, while defaulting to the
+current behaviour.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Rust implements the `env!()` and `option_env!()` macros to access the process environment variables at compilation time.
+`rustc` supports a number of command-line options to control the environment visible to the compiling code.
+
+By default all environment variables are available with their value taken from the environment. There are several
+additional controls to control the logical environment accessed by `env!()`/`option_env!()`:
+- only allow access to a specific whitelist of variables
+- override specific variables to other values
+- add new variables without them being present in the environment
+
+These options are:
+- `--env-whitelist REGEX` - match the REGEX against all existing process environment
+  variables and allow them to be seen. This overrides `--env-remove-all`. The regex is matched against the entire variable name
```

Comment:
It seems that `--env-remove-all` is not defined in the RFC?

Reply:
`--env-blacklist .*`